src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java

```diff
@@ -259,6 +259,116 @@ public void testFileTabCharacter() throws Exception {
+        final String[] expectedOne = {
+            "19:25: Line contains a tab character.",
+        };
+        testByConfig(defaultContextConfigPathOne, inputFile, expectedOne);
+        testByConfig(zeroContextConfigPathOne, inputFile, expectedOne);
+
+        final String defaultContextConfigPathTwo =
+                "strategy/FileTabCharacter/MoveCodeInSameFileWithoutChanges"
+                        + "/patchedline/defaultContextConfig.xml";
+        final String zeroContextConfigPathTwo =
+                "strategy/FileTabCharacter/MoveCodeInSameFileWithoutChanges/"
+                        + "patchedline/zeroContextConfig.xml";
+        final String[] expectedTwo = {
+            "19:25: Line contains a tab character.",
+        };
+        testByConfig(defaultContextConfigPathTwo, inputFile, expectedTwo);
+        testByConfig(zeroContextConfigPathTwo, inputFile, expectedTwo);
+    }
+
+    @Test
+    public void testFileTabCharacterMoveCodeInSameFileWithMinorChanges() throws Exception {
+        final String inputFile =
+                "strategy/FileTabCharacter/MoveCodeInSameFileWithMinorChanges/Test.java";
+
+        final String defaultContextConfigPathOne =
+                "strategy/FileTabCharacter/"
+                        + "MoveCodeInSameFileWithMinorChanges/newline/defaultContextConfig.xml";
+        final String zeroContextConfigPathOne =
+                "strategy/FileTabCharacter/"
+                        + "MoveCodeInSameFileWithMinorChanges/newline/zeroContextConfig.xml";
+        final String[] expectedOne = {
+            "19:25: Line contains a tab character.",
+        };
+        testByConfig(defaultContextConfigPathOne, inputFile, expectedOne);
+        testByConfig(zeroContextConfigPathOne, inputFile, expectedOne);
+
+        final String defaultContextConfigPathTwo =
+                "strategy/FileTabCharacter/MoveCodeInSameFileWithMinorChanges"
+                        + "/patchedline/defaultContextConfig.xml";
+        final String zeroContextConfigPathTwo =
+                "strategy/FileTabCharacter/MoveCodeInSameFileWithMinorChanges"
+                        + "/patchedline/zeroContextConfig.xml";
+        final String[] expectedTwo = {
+            "19:25: Line contains a tab character.",
+        };
+        testByConfig(defaultContextConfigPathTwo, inputFile, expectedTwo);
+        testByConfig(zeroContextConfigPathTwo, inputFile, expectedTwo);
+    }
+
+    @Test
+    public void testFileTabCharacterSingleChangedLine() throws Exception {
+        final String inputFile = "strategy/FileTabCharacter/SingleChangedLine/Test.java";
+
+        final String defaultContextConfigPathOne =
+                "strategy/FileTabCharacter/SingleChangedLine/newline/defaultContextConfig.xml";
+        final String zeroContextConfigPathOne =
+                "strategy/FileTabCharacter/SingleChangedLine/newline/zeroContextConfig.xml";
+        final String[] expectedOne = {
+            "Audit done.",
```

Comment:
Only violations are expected here. If there are no violations just use an empty array. This should avoid that logic below you used:
```
if (!"Audit done.".equals(expected[i])) {....
...
```

Reply:
done